### PR TITLE
AddNParticles: Fix SoA assignment for Momentum

### PR DESCRIPTION
**Pull request by cemitch99**
The debugging changes that we discussed earlier.

### CODE REVIEW

**Comment by ax3l** [comment]
I am looking at a funny segfault now in lines
```C++
using PinnedTile = amrex::ParticleTile<NStructReal, NStructInt, NArrayReal, NArrayInt,
                amrex::PinnedArenaAllocator>;
PinnedTile pinned_tile;
```

let's see if I can spot it, seems from copying an empty source tile in `amrex::copyParticles` during allocation, while trying to iterate its empty components.

**Comment by ax3l** [comment]
I am not sure how  this can happen, but I think the line in the debug backtrace line info are off. It's definitely in the final `amrex::copyParticles` call.

File: src/particles/ImpactXParticleContainer.cpp [hunk 1]
```diff
@@ -79,9 +83,6 @@ namespace impactx
         // write Real attributes (SoA) to particle initialized zero
         DefineAndReturnParticleTile(0, 0, 0);
 
-        pinned_tile.push_back_real(RealSoA::ux, np, 0.0);
-        pinned_tile.push_back_real(RealSoA::uy, np, 0.0);
-        pinned_tile.push_back_real(RealSoA::pt, np, 0.0);
         pinned_tile.push_back_real(RealSoA::ux, *px.cbegin(), *px.cend());
         pinned_tile.push_back_real(RealSoA::uy, *py.cbegin(), *py.cend());
         pinned_tile.push_back_real(RealSoA::pt, *pz.cbegin(), *pz.cend());
```
Comment (ax3l) [inline src/particles/ImpactXParticleContainer.cpp]: This did hit the wrong overload of `push_back_real` (was `int` + `value`, should have been pointers):
```suggestion
        pinned_tile.push_back_real(RealSoA::ux, &(*px.cbegin()), &(*px.cend()));
        pinned_tile.push_back_real(RealSoA::uy, &(*py.cbegin()), &(*py.cend()));
        pinned_tile.push_back_real(RealSoA::pt, &(*pz.cbegin()), &(*pz.cend()));
```

**Review by cemitch99 — COMMENTED** [review]
Ok.  Looks like a simple fix.

File: src/particles/ImpactXParticleContainer.cpp (hunk 1 above)
Comment (ax3l) [reply src/particles/ImpactXParticleContainer.cpp]: Will simplify later on to
```suggestion
        pinned_tile.push_back_real(RealSoA::ux, px);
        pinned_tile.push_back_real(RealSoA::uy, py);
        pinned_tile.push_back_real(RealSoA::pt, pz);
```
when https://github.com/AMReX-Codes/amrex/pull/2634 is merged.

Comment (ax3l) [reply src/particles/ImpactXParticleContainer.cpp]: Simplification in #79

**Pull request merged**